packages/clerk-js/src/ui/signUp/utils.ts

```diff
@@ -45,5 +70,75 @@ export function determineFirstPartyFields(environment: EnvironmentResource, hasT
     fields.password = 'required';
   }
 
+  // Enable fields based on current form state
+
+  const shouldShowEmailAddressField =
+    (isEmailAddressUsedForIdentification && hasTicket && hasEmail) ||
+    fields.emailAddress ||
+    (fields.emailOrPhone && activeCommIdentifierType === 'emailAddress');
+
+  if (shouldShowEmailAddressField) {
+    fields.emailAddress ||= 'required'; // Is this accurate?
+  }
+
+  const shouldShowPhoneNumberField =
+    fields.phoneNumber || (fields.emailOrPhone && activeCommIdentifierType === 'phoneNumber');
+
+  if (shouldShowPhoneNumberField) {
+    fields.phoneNumber ||= 'required'; // Is this accurate?
+  }
+
+  // If continuing with an existing sign-up, show only fields absolutely necessary to minimize fiction
+
+  if (signUp) {
+    const hasVerifiedEmail = signUp.verifications?.emailAddress?.status == 'verified';
+    const hasVerifiedPhone = signUp.verifications?.phoneNumber?.status == 'verified';
+    const hasVerifiedExternalAccount = signUp.verifications?.externalAccount?.status == 'verified';
+
+    if (hasVerifiedEmail) {
+      delete fields.emailAddress;
+      delete fields.emailOrPhone;
+    }
+
+    if (hasVerifiedPhone) {
+      delete fields.phoneNumber;
+      delete fields.emailOrPhone;
+    }
+
+    if (hasVerifiedExternalAccount) {
+      delete fields.password;
+    }
+
+    if (signUp.firstName) {
+      delete fields.firstName;
+    }
+
+    if (signUp.lastName) {
+      delete fields.lastName;
+    }
+
+    if (signUp.username) {
+      delete fields.username;
+    }
+
+    // Remove any non-required fields
+    Object.entries(fields).forEach(([k, v]) => {
+      if (v !== 'required') {
+        // @ts-ignore
+        delete fields[k];
+      }
+    });
+  }
+
   return fields;
 }
+
+export function showFormFields(userSettings: UserSettingsResource): boolean {
+  const { socialProviderStrategies, web3FirstFactors } = userSettings;
```

Comment:
❓ 
Do we want `web3FirstFactors` as unused ?

Reply:
Fixing